### DISCUSSION TITLE
product variant price entety has channel id but not related by foreign key why?

### DISCUSSION THREAD

**Discussion by loveyandex** [vendurehq/vendure#1174] [Q&A]
it should be related to channel why not connected with fk??

i understand each productvarient in each  channel must able to have a price for many reasons
so it has to connect channel with foreign key 
in this such way we add it  in language level but not in table relation

![image](https://user-images.githubusercontent.com/53381292/137943017-43d84d37-1e7f-44be-9db1-5a4adf3798ae.png)

create table product_variant_price
(
    "createdAt" timestamp default now() not null,
    "updatedAt" timestamp default now() not null,
    price       integer                 not null,
    id          serial                  not null
        constraint "PK_ba659ff2940702124e799c5c854"
            primary key,
    "channelId" integer                 not null,
    "variantId" integer
        constraint "FK_e6126cd268aea6e9b31d89af9ab"
            references product_variant
);

**Comment by michaelbromley** [comment]
I can't remember the reasoning for this, or maybe there was no reason and it was just an oversight on my part.

**Reply by loveyandex** [reply]
so it says each sell must be in a channel ?
channel  
it is necessery we must define channel at least one?
